File: framework/src/main/java/com/software/dm/swallow/stormy/algoac/inter/State.java
package com.software.dm.swallow.stormy.algoac.inter;

import java.util.Collection;

/**
 * @param <T>
 * @param <D>
 * @author DM
 */
public interface State<T, D> {

    public D nextState(T c);

    public D addState(T c);

    public D nextStateIgnoreRootState(T c);

    public Collection<T> getValues();

    public Collection<D> getStates();

}

File: framework/src/main/scala/com/software/dm/swallow/stormy/scala/algoac/inter/State.scala
package com.software.dm.swallow.stormy.scala
package algoac
package inter


import scala.collection.Iterable

/**
  * @author darlingming@126.com
  * @tparam T
  * @tparam D
  */
trait State[T, D] {
  def nextState(c: T): D

  def addState(c: T): D

  def nextStateIgnoreRootState(c: T): D

  def getValues: Iterable[T]

  def getStates: Iterable[D]
}